### PR TITLE
Msccl v2.19 integrate (#40)

### PR DESCRIPTION
**Pull request by Andyli1007**
1. catch up the latest nccl version
2. merge the msccl api with nccl api in collectives.cc
3. keep the msccl kernel seperate as temperate solution, so can not support compile by reduce function type and operator, etc currently.

### CODE REVIEW

**Comment by yzygitzh** [comment]
Can you help remove cgmanifest.json in root directory? Seems not used.

**Comment by Andyli1007** [comment]
> Can you help remove cgmanifest.json in root directory? Seems not used.

this is required by azure open source requirement.

File: src/device/onerank.cu [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+  void const* kernel;
+  switch (eltType) {
+  case ncclInt8:     kernel = (void const*)&oneRankReduce<FuncPreMulSum<int8_t>>; break;
+  case ncclUint8:    kernel = (void const*)&oneRankReduce<FuncPreMulSum<uint8_t>>; break;
+  case ncclInt32:    kernel = (void const*)&oneRankReduce<FuncPreMulSum<int32_t>>; break;
+  case ncclUint32:   kernel = (void const*)&oneRankReduce<FuncPreMulSum<uint32_t>>; break;
+  case ncclInt64:    kernel = (void const*)&oneRankReduce<FuncPreMulSum<int64_t>>; break;
+  case ncclUint64:   kernel = (void const*)&oneRankReduce<FuncPreMulSum<uint64_t>>; break;
+  case ncclFloat16:  kernel = (void const*)&oneRankReduce<FuncPreMulSum<half>>; break;
+  #if defined(__CUDA_BF16_TYPES_EXIST__)
```
Comment (yzygitzh) [inline src/device/onerank.cu:67]: no indenting for #if/#else/#endif macro?

**Review by yzygitzh — APPROVED** [review]

File: src/device/onerank.cu (hunk 1 above)
Comment (Andyli1007) [reply src/device/onerank.cu]: removed the indenting

**Pull request merged**